src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.netcoreapp.cs

```diff
@@ -20,5 +23,92 @@ public ReadOnlySpan<T> AsSpan(Range range)
             (int start, int length) = range.GetOffsetAndLength(self.Length);
             return new ReadOnlySpan<T>(self.array, start, length);
         }
+
+        /// <summary>
+        /// Removes the specified values from this list.
+        /// </summary>
+        /// <param name="items">The items to remove if matches are found in this list.</param>
+        /// <param name="equalityComparer">
+        /// The equality comparer to use in the search.
+        /// </param>
+        /// <returns>
+        /// A new list with the elements removed.
+        /// </returns>
+        public ImmutableArray<T> RemoveRange(ReadOnlySpan<T> items, IEqualityComparer<T>? equalityComparer = null)
+        {
+            var self = this;
+            self.ThrowNullRefIfNotInitialized();
+
+            if (items.IsEmpty || self.IsEmpty)
+            {
+                return self;
+            }
+
+            if (items.Length == 1)
+            {
+                return self.Remove(items[0], equalityComparer);
+            }
+
+#nullable disable
+            var itemsMultiSet = new Dictionary<T, int>(equalityComparer);
+#nullable restore
+            int nullValueCount = 0;
+            foreach (ref readonly T item in items)
+            {
+                if (item == null)
+                {
+                    nullValueCount++;
+                }
+                else
+                {
+#nullable disable
+                    ref int count = ref CollectionsMarshal.GetValueRefOrAddDefault(itemsMultiSet, item, out _);
+#nullable restore
+                    count++;
+                }
+            }
+
+            List<int>? indicesToRemove = null;
+            T[] selfArray = self.array!;
+            for (int i = 0; i < selfArray.Length; i++)
+            {
+                bool found = false;
+                if (selfArray[i] == null)
+                {
+                    if (nullValueCount > 0)
+                    {
+                        found = true;
```

Comment:
It seems like you could replace the `found` variable with use of `continue`
```C#
if (selfArray[i] is null && nullValueCount-- == 0)
{
    continue;
}
```

Reply:
Sorry maybe not get the point. The purpose of `found` is to avoid duplicated code inside each iteration.
We may need to handle `null` value case and add their indices to indicesToRemove if meet, `continue `way will not help on that and nullValueCount-- inside condition statement may make unnecessary calculation when candidates is empty ?

Reply:
> nullValueCount-- inside condition statement may make unnecessary calculation when candidates is empty ?

I don't have a strong opinion -- it's micro-optimization that likely doesn't matter very much. You might want to make a decision by taking a look at the generated assembly code.

> Sorry maybe not get the point.

Maybe a full example might illustrate what I mean:

```C#
for (int i = 0; i < selfArray.Length; i++)
{
    if (selfArray[i] is null)
    {
        if (nullValueCount == 0)
        {
            continue;
        }

        nullValueCount--;
    }
    else
    {
#nullable disable
        ref int count = ref CollectionsMarshal.GetValueRefOrNullRef(itemsMultiSet, selfArray[i]);
#nullable restore
        // This avoids decrementing the counter when not needed but we're still reading a reference twice. Check generated assembly to find best approach.
        if (Unsafe.IsNullRef(ref count) || count == 0)
        {
            continue;
        }

        count--;
    }

    indicesToRemove ??= new();
    indicesToRemove.Add(i);
}
```

Reply:
Fixed. 
Got the point now, yes it is better, thanks.